staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/structural.go

```diff
@@ -20,6 +20,23 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+const (
+	XListTypeMap    = "map"
+	XListTypeAtomic = "atomic"
+	XListTypeSet    = "set"
+	XMapType        = "granular"
```

Comment:
XMapTypeAtomic

Reply:
Please a string type alias to make this type safe.

Reply:
unless there're strong objections I'd prefer to do this in a different follow up PR so we also update the internal types in `Extensions struct` to the alias